Approved/implicitly-scoped-qubit-allocation.md

```diff
@@ -0,0 +1,347 @@
+
+Both options seem similar in terms of complexity and implementation timeline, but option 2 seems like the cleaner solution.
+Neither option has any performance differences compared to the existing implementation.
+
+## Timeline
+
+There are no dependencies on other proposals or libraries.
+The work needed for either option appears to be minimal, since it is similar to the existing implementation.
+
+# Further Considerations
+
+Development tools should consider adding support for transitioning from the current syntax to the proposed syntax, such as a code action in IDEs.
+
+## Related Mechanisms
+
+This proposal relies on the existing qubit management functionality that is already a core feature of Q#.
+It does not change the behavior of qubit management; it only provides new syntax for existing functionality.
+
+## Impact on Existing Mechanisms
+
+The new keywords `use` and `borrow` were previously valid Q# identifiers.
+Adding them is a breaking change, unless an opt-in mechanism for new keywords is added to the language.
+
+The scope statement added in this proposal can also be used to limit the scope of non-qubit variable bindings declared with `let` and `mutable`.
+This may be useful for limiting the lifetime of large data structures like arrays or for otherwise communicating the intent that a variable should only be used in a specific section within a callable.
+
+## Anticipated Interactions with Future Modifications
+
+Because the proposed modification is similar to existing functionality, it is not expected to have any impact on future language development.
+
+## Alternatives
+
+### Alternative 1: Add new statements while preserving existing block syntax
+
+Q# could add non-block `using` and `borrowing` statements while preserving existing syntax:
+
+```qsharp
+using q = Qubit();
+// ...
+
+using (q = Qubit()) {
+    // ...
+}
+
+borrowing q = Qubit();
+// ...
+
+borrowing (q = Qubit()) {
+    // ...
+}
+```
+
+### Alternative 2: Add compound using and borrowing block statements
+
+Instead of adding non-block statements, Q# could extend the existing block statement syntax to allow multiple `using` and `borrowing` statements in a row that have only one block attached:
+
+```qsharp
+using (a = Qubit())
+borrowing (b = Qubit())
+using (c = Qubit()) {
+    // ...
+}
+```
+
+## Comparison to Alternatives
+
+### Alternative 1
+
+This alternative has the advantage of being backwards-compatible, but introduces inconsistencies and added complexity:
+
+1. The `using` and `borrowing` keywords are re-used in different contexts: sometimes it is a block statement and sometimes it is a non-block statement.
+   New keywords `use` and `borrow` could be introduced for the non-block statements, but this adds redundancy and keyword clutter.
+2. The block statement requires parentheses around the declaration, but the non-block statement does not.
```

Comment:
I agree that dropping the parens would be good if we keep the block version. In that case, we should also drop the requirement for parenthesis after keywords for other block statements (if-, repeat-, while-, and for-statement), and list it under impact on existing mechanisms.

Reply:
The proposal removes parentheses from all block statement headers now.